addon/components/base/bs-form.js

```diff
@@ -220,26 +246,40 @@ export default Component.extend({
    * @private
    */
   submit(e) {
+    this.set('isSubmitting', true);
+
     if (e) {
       e.preventDefault();
     }
     let model = this.get('model');
 
     this.get('onBefore')(model);
 
-    if (!this.get('hasValidator')) {
-      return this.get('onSubmit')(model);
-    } else {
-      let validationPromise = this.validate(this.get('model'));
-      if (validationPromise && validationPromise instanceof RSVP.Promise) {
-        validationPromise
-          .then((r) => this.get('onSubmit')(model, r))
-          .catch((err) => {
-            this.set('showAllValidations', true);
-            return this.get('onInvalid')(model, err);
-          });
-      }
+    let validationPromise = this.get('hasValidator') ? this.validate(this.get('model')) : RSVP.resolve();
+    if (!isPromise(validationPromise)) {
+      validationPromise = RSVP.resolve();
     }
+    validationPromise
```

Comment:
Could also return a Promise from `submit()` but not quite sure what that would be usefull for. It also hasn't returned anything before.